src/cli/crd/generate.ts

```diff
@@ -0,0 +1,316 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023-Present The Pepr Authors
+
+import { Command } from "commander";
+import fs from "fs";
+import path from "path";
+import { stringify as toYAML } from "yaml";
+import {
+  Project,
+  InterfaceDeclaration,
+  TypeAliasDeclaration,
+  SyntaxKind,
+  Node,
+  SourceFile,
+  Type,
+} from "ts-morph";
+import { createDirectoryIfNotExists } from "../../lib/filesystemService";
+import { kind as k } from "kubernetes-fluent-client";
+import { V1JSONSchemaProps } from "@kubernetes/client-node";
+
+export default new Command("generate")
+  .description("Generate CRD manifests from TypeScript definitions")
+  .option("--output <output>", "Output directory for generated CRDs", "./crds")
+  .action(generateCRDs);
+
+export function extractCRDDetails(
+  content: string,
+  sourceFile: SourceFile,
+): {
+  kind: string | undefined;
+  fqdn: string;
+  scope: "Cluster" | "Namespaced";
+  plural: string;
+  shortNames?: string[];
+} {
+  const kind = extractSingleLineComment(content, "Kind");
+  const group = extractSingleLineComment(content, "Group") ?? "example";
+  const domain = extractSingleLineComment(content, "Domain") ?? "pepr.dev";
+  const details = extractDetails(sourceFile);
+
+  const fqdn = `${group}.${domain}`;
+
+  const { plural, scope } = details;
+
+  const shortNames = details.shortName ? [details.shortName] : undefined;
+  return { kind, plural, scope, shortNames, fqdn };
+}
+
+export async function generateCRDs(options: { output: string }): Promise<void> {
```

Comment:
I personally am wary of `Promise<void>` as a return type. Should we return something instead? Perhaps the filepath to where resources were created or some other success indicator?

Maybe something along the lines of `Promise<{ status: 'success', filePath?: string }>`?

Reply:
In this case it is commander that calls this function through an action. I don't really know if this is going to make a difference? idk though, given that do you still feel that way?

```ts
export default new Command("generate")
  .description("Generate CRD manifests from TypeScript definitions")
  .option("--output <output>", "Output directory for generated CRDs", "./crds")
  .action(generateCRDs);
 ```

Reply:
Yeah, this is fine for now. I think I ought to consider the pros/cons of `Promise<void>` in the context of the whole project instead of this within this one PR. We can leave it as-is for now.